controller_interface/include/controller_interface/controller_interface_base.hpp

```diff
@@ -71,6 +71,15 @@ struct ControllerUpdateStats
   unsigned int total_triggers;
   unsigned int failed_triggers;
 };
+
+struct ControllerUpdateStatus
+{
+  bool ok = true;
+  return_type result = return_type::OK;
+  std::optional<std::chrono::nanoseconds> execution_time = std::nullopt;
+  std::optional<rclcpp::Duration> period = std::nullopt;
+};
```

Comment:
```suggestion
struct ControllerUpdateStatus
{
  return_type result = return_type::OK;
  std::optional<std::chrono::nanoseconds> execution_time = std::nullopt;
  std::optional<rclcpp::Duration> period = std::nullopt;
  
  bool operator bool() const { return result == return_type::OK; }
};
```
how about something like this?

Reply:
I sharing the part of code here for reference:
https://github.com/ros-controls/ros2_control/blob/faf9c7596c3b5683bdb9a3c4cc762395f5395751/controller_interface/src/controller_interface_base.cpp#L171-L179

Reply:
ah I see, perhaps we could rename it to avoid confusion later?

Reply:
@bmagyar Done 👍🏾